src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -645,9 +684,15 @@ private void model_PropertyChanged(object sender, PropertyChangedEventArgs e)
                 case "ShowWhitespaceIsChecked":
                 case "NodeAutocompleteIsChecked":
                 case "EnableTSplineIsChecked":
-                    UpdateSavedChangesLabel();
-                    break;
+                    description = Res.PreferencesViewEnableTSplineNodes;
+                    goto default;
                 default:
+                    // Log switch on each setting and use description equals to label name
+                    Dynamo.Logging.Analytics.TrackEvent(
+                        Actions.Switch,
+                        Categories.Preferences,
+                        description);
```

Comment:
This way of logging is more code efficient.. But can't seem to rule out the cases for dialog initialization, because in those cases Dynamo raise property changes regardless if it's user change.. Will look a bit more on best way

Reply:
I take it back, since `Model_PropertyChanged` bind after dialog initialization, this approach should be accurate